apps/web/components/CategoryTree/CategoryTree.vue

```diff
@@ -1,32 +1,44 @@
 <template>
   <div
-    class="py-2 px-4 mb-4 bg-neutral-100 typography-headline-6 font-bold text-neutral-900 uppercase tracking-widest md:rounded-md"
-    data-testid="category-tree"
+    v-if="parent || (categoryTreeItem && categoryTreeGetters.getItems(categoryTreeItem)?.length)"
+    class="category-tree"
   >
-    {{ $t('category') }}
+    <div
+      class="py-2 px-4 mb-4 bg-neutral-100 typography-headline-6 font-bold text-neutral-900 uppercase tracking-widest md:rounded-md"
+      data-testid="category-tree"
+    >
+      {{ $t('category') }}
+    </div>
+    <template v-if="parent">
+      <CategoryTreeItem
+        :name="categoryTreeGetters.getName(parent)"
+        :href="categoryTreeGetters.generateCategoryLink(categoryTree, parent)"
+        :count="categoryTreeGetters.getCount(parent)"
+      >
+        <SfIconArrowBack size="sm" class="text-neutral-500 mr-2" />
+      </CategoryTreeItem>
+    </template>
+    <ul v-if="categoryTreeItem" class="mt-4 mb-6 md:mt-2" data-testid="categories">
+      <CategoryTreeItem
+        v-for="(categoryItem, index) in categoryTreeGetters.getItems(categoryTreeItem)"
+        :key="index"
+        :name="categoryTreeGetters.getName(categoryItem)"
+        :href="categoryTreeGetters.generateCategoryLink(categoryTree, categoryItem)"
+        :count="categoryTreeGetters.getCount(categoryItem)"
+      />
+    </ul>
   </div>
-  <template v-if="parent">
-    <CategoryTreeItem :name="parent.name" :href="parent.href" :count="parent.count">
-      <SfIconArrowBack size="sm" class="text-neutral-500 mr-2" />
-    </CategoryTreeItem>
-  </template>
-  <ul class="mt-4 mb-6 md:mt-2" data-testid="categories">
-    <CategoryTreeItem
-      v-for="(category, index) in categories"
-      :key="index"
-      :name="categoryTreeGetters.getName(category)"
-      :href="categoryTreeGetters.generateCategoryLink(categoryTree, category)"
-      :count="categoryTreeGetters.getCount(category)"
-    />
-  </ul>
 </template>
 
 <script setup lang="ts">
 import { categoryTreeGetters } from '@plentymarkets/shop-sdk';
 import { SfIconArrowBack } from '@storefront-ui/vue';
-import type { CategoryTreeProps } from '~/components/CategoryTree/types';
 
 const { data: categoryTree } = useCategoryTree();
+const { data: productsCatalog } = useProducts();
 
-defineProps<CategoryTreeProps>();
```

Comment:
If I want to use the categorytree component, I now have to call the useProducts to change the selected category. That's a bit unintuitive. We might want to have a chat about that